### PR TITLE
(1.16) Cleanup and optimise AI

### PR DESCRIPTION
**Pull request by PaintNinja**
- Add missing `@Override`s
- Improved style consistency
- Implemented a cache for lists that call the expensive AABB to improve performance
    - This exploits the likelihood that players won't move far within 0.05s (1 tick), instead caching and reusing the calculated list for up-to 4 ticks
    - The cache is updated every 4th tick and cleared when no longer in use to prevent stale data.
- Optimised maths code, marked primitives as final where appropriate
- Reduced unnecessary calculations and checks, commented out dead code
- Replaced `new Random()` with `ThreadLocalRandom.current()`
- Changed size checks from `slotlist.size() >= 1` to the faster and clearer `slotlist.isEmpty()`

### CODE REVIEW

File: src/main/java/com/github/alexthe666/iceandfire/entity/ai/MyrmexAIAttackPlayers.java [hunk 1]
```diff
@@ -1,25 +1,30 @@
...
 
+    @SuppressWarnings("unchecked")
     public MyrmexAIAttackPlayers(EntityMyrmexBase myrmex) {
         super(myrmex, PlayerEntity.class, 10, true, true, new Predicate<PlayerEntity>() {
-            public boolean apply(@Nullable PlayerEntity entity) {
-                return entity != null && (myrmex.getHive() == null || myrmex.getHive().isPlayerReputationLowEnoughToFight(entity.getUniqueID()));
+
+            @Override
+            public boolean test(PlayerEntity entity) {
+                return (myrmex.getHive() == null
```
Comment (TheBv) [inline src/main/java/com/github/alexthe666/iceandfire/entity/ai/MyrmexAIAttackPlayers.java]: Consider adding a null check for the player

File: src/main/java/com/github/alexthe666/iceandfire/entity/ai/PixieAIPickupItem.java [hunk 2]
```diff
@@ -41,9 +39,11 @@ public PixieAIPickupItem(EntityPixie creature, int chance, boolean checkSight, b
 
         this.targetEntitySelector = new Predicate<ItemEntity>() {
             @Override
-            public boolean apply(@Nullable ItemEntity item) {
+            public boolean test(ItemEntity item) {
 
-                return item != null && !item.getItem().isEmpty() && (item.getItem().getItem() == Items.CAKE && !creature.isTamed() || item.getItem().getItem() == Items.SUGAR && creature.isTamed() && creature.getHealth() < creature.getMaxHealth());
+                return !item.getItem().isEmpty() && (item.getItem().getItem() == Items.CAKE && !creature.isTamed()
```
Comment (TheBv) [inline src/main/java/com/github/alexthe666/iceandfire/entity/ai/PixieAIPickupItem.java]: Same here

**Comment by TheBv** [comment]
Looks good :)
Thanks for your work!

File: src/main/java/com/github/alexthe666/iceandfire/entity/ai/MyrmexAIAttackPlayers.java (hunk 1 above)
Comment (Matyrobbrt) [reply src/main/java/com/github/alexthe666/iceandfire/entity/ai/MyrmexAIAttackPlayers.java]: Done

File: src/main/java/com/github/alexthe666/iceandfire/entity/ai/PixieAIPickupItem.java (hunk 2 above)
Comment (PaintNinja) [reply src/main/java/com/github/alexthe666/iceandfire/entity/ai/PixieAIPickupItem.java]: Done here https://github.com/Alex-the-666/Ice_and_Fire/pull/4470/files#diff-139982523db614398c7410078f2b8f1d8f3149272d7349ca76b453afda0f7434R48

Comment (TheBv) [reply src/main/java/com/github/alexthe666/iceandfire/entity/ai/PixieAIPickupItem.java]: Oh whoops didn't catch that sorry!

**Pull request merged**